### PR TITLE
Add __len__ for DistributedWeightedRandomSampler

### PR DESCRIPTION
**Pull request by Nic-Ma**
### Description
This PR added the `__len__()` API for `DistributedWeightedRandomSampler`, refer to the PyTorch source code:
https://github.com/pytorch/pytorch/blob/master/torch/utils/data/sampler.py#L189

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/data/samplers.py [hunk 1]
```diff
@@ -107,16 +107,18 @@ def __init__(
...
         weights = torch.as_tensor([self.weights[i] for i in indices], dtype=torch.double)
         # sample based on the provided weights
-        rand_tensor = torch.multinomial(weights, num_samples, self.replacement, generator=self.generator)
+        rand_tensor = torch.multinomial(weights, self.num_samples_per_rank, self.replacement, generator=self.generator)
 
         for i in rand_tensor:
             yield indices[i]
+
+    def __len__(self):
+        return self.num_samples_per_rank
```
Comment (wyli) [inline monai/data/samplers.py:124]: not sure how we achieve `replacement=False` in a distributed setting, could you add a test for https://github.com/pytorch/pytorch/blob/master/torch/utils/data/sampler.py#L164 or we could remove the `False` option...

**Review by wyli — APPROVED** [review]
thanks!

File: monai/data/samplers.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/data/samplers.py:119]: Sounds reasonable.
Removed `replacement` arg in the latest commit.

Thanks.

**Pull request merged**